InTheNewsNotification/WMFInTheNewsNotificationViewController.swift

```diff
@@ -134,11 +134,14 @@ class WMFInTheNewsNotificationViewController: ExtensionViewController, UNNotific
         case WMFInTheNewsNotificationSaveForLaterActionIdentifier:
             statusView.isHidden = false
             statusLabel.text = WMFLocalizedString("status-saving-for-later", value:"Saving for later...", comment: "Indicates to the user that the article is being saved for later")
-            let dataStore = MWKDataStore.shared()
-            dataStore.savedPageList.addSavedPage(with: articleURL)
-            self.statusView.isHidden = false
-            self.statusLabel.text = WMFLocalizedString("status-saved-for-later", value:"Saved for later", comment: "Indicates to the user that the article has been saved for later")
-            completion(.dismiss)
+            WidgetController.shared.startWidgetUpdateTask(completion) { (dataStore, completion) in
+                DispatchQueue.main.async {
+                    dataStore.savedPageList.addSavedPage(with: articleURL)
+                    self.statusView.isHidden = false
+                    self.statusLabel.text = WMFLocalizedString("status-saved-for-later", value:"Saved for later", comment: "Indicates to the user that the article has been saved for later")
+                    completion(.dismiss)
```

Comment:
Some more completion confusion on my end, seems like you should call the inner completion to release the `dataStore` (the one returned next to `dataStore` in line 137), it's unclear if that would happen.

Reply:
The variable name is shaded here so they're both `completion` - I'll rename it for clarity